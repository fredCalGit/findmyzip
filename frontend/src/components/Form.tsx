import * as React from "react";
import TextField from "@mui/material/TextField";
import Autocomplete from "@mui/material/Autocomplete";
import countries from "../utils/countriesList";
import { Box, Button } from "@mui/material";
import { SyntheticEvent, useEffect, useState } from "react";

interface FormProps {
  handleSubmit: ({
    country,
    postCode,
  }: {
    country: string;
    postCode: string;
  }) => void;
}
export const Form = ({ handleSubmit }: FormProps) => {
  const [country, setCountry] = useState("United States");
  const [zipCode, setZipCode] = useState("");

  const getCountryAlias = (name: string) => {
    const result = countries.filter(
      (country) => country.label.toLowerCase() === name.toLowerCase()
    );
    return result[0].value;
  };
  const handleCountryChange = (newValue: string) => {
    if (newValue) {
      setCountry(newValue);
    }
  };

  const handleZipChange = (newValue: string) => {
    setZipCode(newValue);
  };

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-around",
      }}
    >
      <Autocomplete
        disablePortal
        id="country-input"
        options={countries
          .map((country) => {
            return country.label;
          })
          .sort((a, b) => -b)}
        autoHighlight
        sx={{ width: 300 }}
        renderInput={(params) => (
          <TextField {...params} label="Choose a Country" />
        )}
        value={country}
        inputValue={country}
        onChange={(event, value) => {
          if (value) handleCountryChange(value);
        }}
      />
      <TextField
        id="zipcode-input"
        label="Enter Zip-Code"
        variant="outlined"
        sx={{ width: 300 }}
        value={zipCode}
        onChange={(event) => handleZipChange(event.target.value)}
      />
      <Button
        variant="outlined"
        onClick={() => {
          if (zipCode)
            handleSubmit({
              country: getCountryAlias(country),
              postCode: zipCode,
            });
        }}
      >
        Search
      </Button>
    </Box>
  );
};
