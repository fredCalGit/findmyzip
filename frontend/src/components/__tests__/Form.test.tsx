import { render } from "@testing-library/react";
import { Form } from "../Form";

test("renders the Form", () => {
  const mockFunction = jest.fn();

  const { container } = render(<Form handleSubmit={mockFunction} />);
  expect(container).toMatchSnapshot();
});
