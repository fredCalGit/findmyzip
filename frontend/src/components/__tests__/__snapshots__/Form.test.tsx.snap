// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Form 1`] = `
<div>
  <div
    class="MuiBox-root css-1vt08d7"
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-bs2jcm-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="country-input"
          id="country-input-label"
        >
          Choose a Country
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="country-input"
            role="combobox"
            spellcheck="false"
            type="text"
            value="United States"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Choose a Country
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-9ahdq0-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="zipcode-input"
        id="zipcode-input-label"
      >
        Enter Zip-Code
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="zipcode-input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Enter Zip-Code
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
